modules/veteran/spec/models/veteran/service/representative_spec.rb

```diff
@@ -120,6 +120,16 @@ def basic_attributes
 
           expect(representative.reload.full_name).to eq('Bob Smith')
         end
+
+        it 'does not throw an exception when one of the names if nil' do
+          representative = create(:representative, first_name: 'Joe', last_name: nil)
+
+          expect(representative.full_name).to eq('Joe ')
+
+          representative.update(first_name: 'Bob')
+
+          expect(representative.reload.full_name).to eq('Bob ')
+        end
```

Comment:
@stiehlrod  I'm not sure this test is actually hitting that `all_for_user` method at all.  When I run it with a debugger nothing happens there. Am I missing something on that? 

This did hit the method as expected (I added it at line 55)
```
      it 'finds all for a user with nil last name' do
        expect(Veteran::Service::Representative.all_for_user(
          first_name: identity.first_name,
          last_name: nil
        ).id).to eq(representative.id)
      end
```
But I am wondering if the query works without one of those values in it?  It did not return anything with just the first name.

Again if I am missing some context on that let me know.